pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -131,8 +132,10 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 
 		// Update the status.conditions. This only works the first time, so if you fix an issue, it will still be set to PlatformXXXMisConfigured
 		setStatusCondition(&hyd, hypdeployment.PlatformConfigured, metav1.ConditionFalse, "Configuring platform with infra-id: "+hyd.Spec.InfraID, hypdeployment.BeingConfiguredReason)
-		setStatusCondition(&hyd, hypdeployment.WorkConfigured, metav1.ConditionFalse, "Configuring ManifestWork: "+hyd.Spec.InfraID, hypdeployment.BeingConfiguredReason)
-		r.updateStatusConditionsOnChange(&hyd, hypdeployment.PlatformIAMConfigured, metav1.ConditionFalse, "Configuring platform IAM with infra-id: "+hyd.Spec.InfraID, hypdeployment.BeingConfiguredReason)
+		if hyd.Spec.Infrastructure.Platform != nil && hyd.Spec.Infrastructure.Platform.AWS != nil {
```

Comment:
So we need an additional statement to set the IAM condition to unused or NA or does not applied. Otherwise our pretty print columns for `oc get hd` will be blank for IAM on Azure.
Maybe a value of `unused` or `none`